tests/test_utils.py

```diff
@@ -187,6 +188,21 @@ def test_quickSmooth(self):
         self.assertEqual(data.shape, (100, 100))
 
 
+class QuantileTestCase(lsst.utils.tests.TestCase):
+    def setUp(self) -> None:
+        return super().setUp()
+
+    def test_quantiles(self):
+        # We understand that our algorithm gives very large rounding error
+        # compared to the generic numpy method. But still test it.
+        np.random.seed(1234)
+        data = np.random.normal(100, 10, 100)
```

Comment:
I think it might be worth having some 2d data in here (to make sure we don't break the ravelling) and maybe something with some bigger values, to make sure things will work when we have values up to 2^18 (which is the number of bits in our ADC).

Reply:
👍  about 2d data. 

I hesitate about much bigger values. Two numbers are also of interest here:
- `(maxVal - minVal)` which is then the histogram range and we fix the bin size to 1 
- `nColors` which we'd just use 256 
Larger values means larger imprecision too. Sure we can adjust the test to allow that.  But I'm not sure how meaningful it really is. 

Thoughts? How about the updated numbers in the test?

Reply:
Rather than iterating in the comments here, I just pushed a commit with the changes I was imagining - I know that's a little irregular, I hope that's OK, obviously feel free to edit it however you like!

I added: several different ranges and numbers of colours, and also set an element in the array to `nan` to check we're nan-safe (and therefore changed the `np.quantile` call to be `np.nanqualite`)

Let me know what you think (and if you don't like me pushing to your branch in general!)

Reply:
Good ideas in testing a variety of parameters, in testing `nan`,  and so on.

The only thing is that I cannot convince myself that data of (mean, width)= (2, 0.1) would be meaningful here. So I adjusted it to be a little bigger, yet small for possible input images I think.    Also I'm adjusting the tolerance slightly depending on the inputs.

Reply:
Most pixels on such image would be approximated into just 1 and 2 with the current implementation. Probably easier if we have some example images. I'll leave it to a future ticket.